util.py

```diff
@@ -0,0 +1,97 @@
+from urllib.parse import unquote
+
+from bs4 import BeautifulSoup
+import fetch
+
+
+def coupon_extract(start_position=0, end_position=0, html=""):
+    coupon_position = html.find("couponCode")
+    if coupon_position == -1:
+        return
+
+    # looking for the first character of the link "
+    for i in range(coupon_position, -1, -1):
+        if html[i] == '"':
+            start_position = i+1
+            break
+
+    # Looking for the last character of the link "
+    end_position = html.find('"', start_position)
+
+    # better we unquote the string
+    link_udemy = unquote(html[start_position:end_position])
```

Comment:
Probably better to parse it, and get the params- `urlparse()`, then get the `.query` and `.path` attributes.